Packs/TOPdesk/Integrations/TOPdesk/TOPdesk.py

```diff
@@ -0,0 +1,1303 @@
+            use_new_query: Whether to use FIQL query or not.
+
+        Return the joined query with the argument (e.g. id==some-id&email==my@email.com)
+        """
+        if filter_name and filter_arg:
+            if query:
+                query = f"{query}&"
+            else:
+                query = ''
+
+            if use_new_query:
+                query = f"{query}{filter_name}=={filter_arg}"
+            else:
+                query = f"{query}{filter_name}={filter_arg}"
+
+        return query
+
+    @staticmethod
+    def add_query_to_request(query: Optional[str], url_suffix: str, new_query: bool,
+                             inline_parameters: bool) -> str:
+        """Add the inline query parameter to the url suffix of a request.
+        Consider the supported query type.
+
+        Args:
+             query: The query argument to add to the url suffix (e.g. id==3)
+             url_suffix: The existing url suffix (e.g. /persons?start=2)
+             new_query: Whether to use FIQL query or add parameters inline.
+             inline_parameters: Whether there are already other inline parameters in the url.
+
+        Return the reconstructed url_suffix containing the query.
+        """
+        if query:
+            if new_query:
+                query = f"query={query}"
+
+            if inline_parameters:
+                url_suffix = f"{url_suffix}&{query}"
+
+            else:
+                url_suffix = f"{url_suffix}?{query}"
+
+        return url_suffix
+
+    @staticmethod
+    def convert_query_types(current_query: Optional[str], to_new_query: bool) -> Optional[str]:
+        """Convert inline params to FIQL query and otherwise
+
+        Args:
+            current_query: The current query, must be in the for.
+            to_new_query: Wether to convert to FIQL query or inline parameters.
+
+        Return the new reconstructed query.
+        """
+        if not current_query:
+            return None
+        query_args = re.split("&", current_query)
+        new_query_args = []
+        for query_arg in query_args:
+            query_arg_list = re.split("=", query_arg)
+            if '' in query_arg_list:
+                query_arg_list.remove('')
+
+            if len(query_arg_list) > 3:
+                raise ValueError('Invalid query, make sure it is in the right format')
+
+            if not to_new_query and len(query_arg_list) != 2:
+                raise ValueError(f'Invalid query, older {INTEGRATION_NAME} versions only support filtering with =')
+
+            if to_new_query and len(query_arg_list) == 2 and query_arg_list[0][-1] != '!':
+                new_query_args.append('=='.join(query_arg_list))
+
+            else:
+                new_query_args.append('='.join(query_arg_list))
+
+        return '&'.join(new_query_args)
+
+
+''' HELPER FUNCTIONS '''
+
+
+def trim_results_by_limit(results: List[Any], limit: Union[int, str] = 100) -> List[Any]:
```

Comment:
the issue with this command is that you still bring all the info locally so you don't save the effort for the large HTTP request - is there no `limit` option from the API itself? (if not then this is good and you can resolve this comment.)